lib/cache/factories/cacheFactory.ts

```diff
@@ -0,0 +1,44 @@
+import { CacheOptions } from '..';
```

Comment:
I consider that all factory implementation shouldn't be separated in a specific folder, but rather be part of their domain like `cache/factories/cacheFactory.ts` -> `cache/caches/factory.ts`